src/renderer/entities/signatory/ui/SelectableSignatory/SelectableSignatory.tsx

```diff
@@ -1,51 +1,63 @@
-import { AccountAddress, getAddress, AccountAddressProps } from '@renderer/entities/wallet';
-import { useAddressInfo } from '@renderer/entities/wallet/lib/useAddressInfo';
-import { InfoPopover, Icon } from '@renderer/shared/ui';
-import { toAccountId, transferableAmount, cnTw } from '@renderer/shared/lib/utils';
-import { useBalance, AssetBalance } from '@renderer/entities/asset';
-import type { Explorer, Asset, ChainId } from '@renderer/shared/core';
+import { useUnit } from 'effector-react';
+
+import { WalletIcon, walletModel } from '@renderer/entities/wallet';
+import { Icon } from '@renderer/shared/ui';
+import { cnTw, toAccountId, toAddress, transferableAmount } from '@renderer/shared/lib/utils';
+import { AssetBalance, useBalance } from '@renderer/entities/asset';
+import type { AccountId, Asset, ChainId } from '@renderer/shared/core';
+import { WalletType } from '@renderer/shared/core';
+import { Body } from '@renderer/shared/ui/Typography/Typography.stories';
 
 type Props<T extends any> = {
-  explorers?: Explorer[];
   value: T;
-  onSelected: (value: T) => void;
-  chainId: ChainId;
   asset: Asset;
-} & AccountAddressProps;
+  accountId: AccountId;
+  addressPrefix: number;
+  chainId: ChainId;
+  walletId: number;
+  onSelected: (value: T) => void;
+};
 
 export const SelectableSignatory = <T extends any>({
-  explorers,
-  size = 20,
   value,
-  onSelected,
-  chainId,
   asset,
-  name,
-  ...addressProps
+  accountId,
+  addressPrefix,
+  chainId,
+  walletId,
+  onSelected,
 }: Props<T>) => {
-  const address = getAddress(addressProps);
+  const wallets = useUnit(walletModel.$wallets);
+  const signatoryWallet = wallets.find((w) => w.id === walletId);
+
+  if (!signatoryWallet) return <></>;
+
+  const address = toAddress(accountId, { prefix: addressPrefix });
 
   const { getLiveBalance } = useBalance();
-  const popoverItems = useAddressInfo(address, explorers, true);
   const balance = getLiveBalance(toAccountId(address), chainId, asset.assetId.toString());
 
+  // support for legacy signatory as shard from multishard account
+  const walletType =
+    signatoryWallet.type === WalletType.MULTISHARD_PARITY_SIGNER
+      ? WalletType.SINGLE_PARITY_SIGNER
+      : signatoryWallet.type;
```

Comment:
looks like this code can be used in other places, can you make it as a function?

Reply:
I moved this code into WalletIcon component in other PR because from now on we show legacy multishard wallet as polakadot vault.